SPEC-draft.txt

```diff
@@ -0,0 +1,267 @@
+
+   Any frontend application consists of two parts: a Server and Client
+   Side Assets.
+
+2.1  Server
+
+   The Server is the part of the application to responds to HTTP
+   requests. At a bare minimum the only thing it does is serve the
+   Client Side Assets.
+
+   But it may do a lot more, including, but not limited to, generating
+   HTML pages and other assets on the fly.
+
+   In many frontend applications, the server is a regular static file
+   hosting service that serves a static HTML file.
+
+2.2  Client Side Assets
+
+   The Client Side Assets are the files that contain the code and other
+   assets that run inside the browser such as Javascript or CSS files
+   images or configuration.
+
+
+3.  Design considerations & constraints
+
+   The design of the FAB specification is driven by a list of qualities
+   that we strive for in particular order. Whenever we need to make a
+   trade-off, the higher ordered quality weighs more heavily than lower
+   ordered qualities.
+   Implementors of the specification SHOULD follow the same list of
+   qualities when considering trade-offs.
+   
+   3.1.  Availability & Reliability
+
+   As FABs will be used by applications in production, it is
+   vital that reliability is our absolute top priority.
+   We MUST always strive to optimise for the availability of
+   applications. This implies a few other related qualities, such as
+   security, predictability, resilience and ability to recover from
+   errors.
+
+   It is this reliability that will generate trust in the ecosystem and
+   our deploys, allowing application developers to release more often.
+   This will generate a virtuous cycle of more frequent, less risky
+   releases.
+
+   There MUST NOT be any downtime for regular deploys.
+
+   Most importantly, the functional behaviour of an application MUST
+   be the same when run in different hosting environments with the same
+   settings.
+   A hosting environment MAY have different performance characteristics
+   and other optimisations.
+
+   3.2.  Compatibility
+
+   Because the purpose of Frontend Application bundles is to provide
+   a standard for packaging frontend applications and the runtime 
+   environment they run in, it is important that it is compatible with
+   as many different applications and hosting providers.
+
+   3.2.1.  Application compatibility
+
+   There are many different patterns, frameworks and practices when it
+   comes to frontend applications. Broadly speaking there are three 
+   patterns: "Pre-rendering", "Client-Side Rendering" (CSR) and
+   "Server-Side Rendering" (SSR).
+   And while we should always strive to support as wide a variety of
+   applications, we should optimise for the more dynamic ones, such as 
+   SSR and CSR.
+
+   And while we see great promise in non-Javascript languages on the
+   browser platform (particularly WebAssembly), we will not be
+   officially supporting it. Especially not for the server.
```

Comment:
This is too strong. It might not be explicitly addressed yet, but it will be in the future.

Reply:
I have removed this bit from the application section into the server section. And specified that we will currently only be defining JavaScript/Typescript interfaces.